### PR TITLE
Gui testcases hcs

### PR DESCRIPTION
**Pull request by maryvictol**
This PR add testcases for HCS Analysis.

### CODE REVIEW

File: docs/testcases/UI/HCS/2489_2758.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+- User that has full permissions to the storage with HCS images
+- 2 HCS images (`image1`, `image2`) with the same channels
+
+| Steps | Actions | Expected results |
+| :---: | --- | --- |
+| 1 | Login as the user from the prerequisites | |
+| 2 | Open the **Library** page | |
+| 3 | Find HCS `image1` from the Prerequisites |  |
+| 4 | Click the **HCS** icon near the HCS `image1` file name | ***Preview pop-up*** opens with the global image of the selected well|
+| 5 | Click **Gear** icon on the *Preview panel* | ***Settings*** panel opens that includes <li> *Blending mode:* drop-down <li> *Persist channels state* checkbox <li> All channels list corresponds to the current plate (by default, all channels are selected) <li> For each channel item there are: <ul><li> checkbox to enable the channel <li> label with the channel name <li> the color in which images of that channel are highlighted <li> the horizontal slider of the channel contrast should be placed </ul><li> ***Well view***, ***Field view***, ***Download current image*** buttons |
```
Comment (NShaforostov) [inline docs/testcases/UI/HCS/2489_2758.md]: For each channel, also "lock"-control is added now. It allows to lock the state of the separate channel

File: docs/testcases/UI/HCS/2527.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+- User that has full permissions to the storage with HCS images
+- HCS image with several channels
+
+| Steps | Actions | Expected results |
+| :---: | --- | --- |
+| 1 | Login as the user from the prerequisites | |
+| 2 | Open the **Library** page | |
+| 3 | Find HCS image from the Prerequisites |  |
+| 4 | Click the **HCS** icon near the found file name | ***Preview pop-up*** opens with the global image of the first well|
+| 5 | Click **Download** icon on the *Preview panel* | **Save as** pop-up opens <li> *File name* field is prefilled as `Well <col_row>.png`, where `<col_row>` are coordinates of well |
```
Comment (NShaforostov) [inline docs/testcases/UI/HCS/2527.md]: Probably, it depends on the browser settings, but I see that image is downloaded automatically, without **Save as** pop-up.

File: docs/testcases/UI/HCS/2527.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+| 6 | Click **Save** button | |
+| 7 | Open file stored at step 6 | Stored file corresponds image loaded at the preview form at the moment |
+| 8 | On the *Well panel* select any field | |
+| 9 | Click **Gear** icon on the *Preview panel* | ***Settings*** panel opens |
+| 10 | Untick checkbox near the first channel. Set new contrast range for all other channels: channel boundary values (min and max) | |
+| 11 | Zoom image |
+| 12 | Click ***Download current image*** button on the ***Settings*** panel | **Save as** pop-up opens <li> *File name* field is prefilled as `Well <col_row>, Field <num>.png`, where `<col_row>` are coordinates of well, <num>is number of field |
```
Comment (NShaforostov) [inline docs/testcases/UI/HCS/2527.md]: Probably, it depends on the browser settings, but I see that image is downloaded automatically, without **Save as** pop-up.

File: docs/testcases/UI/HCS/2528.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+| 7 | Untick checkbox near the first channel | |
+| 8 | Set new contrast range for all other channels: channel boundary values (min and max) | |
+| 9 | Select any other well on the *Plate panel* | <li> The global image of the whole well is shown on the Preview panel <li> Checkbox near the first channel is unticked on the ***Settings*** panel <li> All channels have contrast range specified at step 8 |
+| 10 | Click ***Field view*** button on the ***Settings*** panel | <li> The field with the 1st ID in the mapping file is highlighted by orange <li> The image correspond highlighted field openes on the *Preview panel* <li> Checkbox near the first channel is unticked on the ***Settings*** panel <li> All channels have contrast range specified at step 8 |
+| 11 | On the *Well panel* select several fields using *Shift* button | <li> The global image of the whole well with highlighted fields selected on the *Well panel* is shown on the *Preview panel*  <li> Checkbox near the first channel is unticked on the ***Settings*** panel <li> All channels have contrast range specified at step 8  |
+| 12 | Untick the ***Persist channels state*** checkbox  | |
```
Comment (NShaforostov) [inline docs/testcases/UI/HCS/2528.md]: Now, for each channel own "lock"-control is added. It allows to lock the state of the separate channel.
I think, verification of this feature can be added to the current testcase.

File: docs/testcases/UI/HCS/2694_4.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+| 7 | Select pipeline created at step 21 of the [2694_3](2694_3.md) case | **Open** button becomes active | 
+| 8 | Click **Open** button | All sections of the Analysis panel are filled as in the [2694_3](2694_3.md) case |
+| 9 | Click **>** button to submit analysis | **Evaluations** tab opens that contains new job (selected by default) with name corresponding HCS image file name |
+| 10 | Expand **Details** section | **Details** section opens that contains <li> `Owner`, `Started`, `Duration` fields <li> `Job ID` with link to run log page <li> `HCS Image` with link to HCS image file <li> `Output folder` with link to folder where output results are stored <li> `Selection` field with information about `wells`, `fields`, `time points` and `z-planes` selected for analysis |
+| 11 | Wait until job is completed | ***Analysis results*** table appears under the **Details** section that contains <li> `Well`, `Timepoint`, `WellRow`, `WellColumn`, `Plane`, `Number of Fields` columns <li> Columns for properties selected at step 8 with statistics selected at step 9 <li> Columns for results of of applying the formula added at steps 10-13 <li> ***Download*** link |
```
Comment (NShaforostov) [inline docs/testcases/UI/HCS/2694_4.md]: Add phrase `of the [2694_3](2694_3.md) case` to `...selected a step 8...` and `...selected a step 9...`, and `...added at steps 10-13...`

File: docs/testcases/UI/HCS/2749.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+| :---: | --- | --- |
+| 1 | Login as the user from the prerequisites | |
+| 2 | Open the **Library** page | |
+| 3 | Find HCS image from the Prerequisites | |
+| 4 | Click the **HCS** icon near the HCS image file name | ***Preview pop-up*** opens that contains <li> All available image Z-planes are shown in the *Z-planes* section: <ul><li> first Z-plane is selected by default <li> *Select all* link is shown |
+| 5 | Click **Analysis** icon on the *Preview panel* | |
+| 6 | Select any field on the *Well panel* | |
+| 7 | Click *Select all* link | Z-planes panel: <li> All z-planes are selected <li> *Use planes projection* checkbox appears <li> *Select all* link disappears </li> Analysis panel: <li> Message `Selection: 1 well, 1 field, 1 time point, <n> z-planes. Analysis will be submitted as a batch job that can be found at Evaluations tab. For fine-tuning pipeline select a single well, field, and time point.` is shown on the Analysis panel |
+| 8 | Tick *Use planes projection* checkbox | Message about batch job disappears from Analysis panel |
+| 9 | Click 2nd z-plane | <li> 2nd z-plane is unselected <li> *Select all* link appears |
```
Comment (NShaforostov) [inline docs/testcases/UI/HCS/2749.md]: Fix to `Holding Shift key, click 2nd z-plane...`

File: docs/testcases/UI/HCS/2766.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
+# [Manual] Video from images stack
+
+Test verifies that User can view the same field but in separate time points to compare some parts of the image as a short video and download it.
+
+**Prerequisites**:
+- User that has full permissions to the storage with HCS images
+- HCS image (`image1`) with only one timepoint in the image
```
Comment (NShaforostov) [inline docs/testcases/UI/HCS/2766.md]: Add `... and only one Z-plane`

File: docs/testcases/UI/HCS/2766.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+- HCS image (`image2`) with several timepoints in the image
+
+| Steps | Actions | Expected results |
+| :---: |---| --- |
+| 1 | Login as the user from the prerequisites | |
+| 2 | Open the **Library** page | |
+| 3 | Find HCS `image1` from the Prerequisites | |
+| 4 | Click the **HCS** icon near the HCS `image1` file name | ***Preview pop-up*** opens that contains **Video** icon on the *Preview panel* |
```
Comment (NShaforostov) [inline docs/testcases/UI/HCS/2766.md]: > that contains ...

Fix to `that does not contain ...`

File: docs/testcases/UI/HCS/2766.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
+# [Manual] Video from images stack
+
+Test verifies that User can view the same field but in separate time points to compare some parts of the image as a short video and download it.
+
+**Prerequisites**:
+- User that has full permissions to the storage with HCS images
+- HCS image (`image1`) with only one timepoint in the image
+- HCS image (`image2`) with several timepoints in the image
```
Comment (NShaforostov) [inline docs/testcases/UI/HCS/2766.md:10]: Now, the ability to create a video for HCS image with several Z-planes is implemented.
Suggest to add following checks:
- creation of video for HCS image with several Z-planes
- switch of video regimens for HCS images that have several timepoints and several Z-planes simultaneously

**Review by NShaforostov — COMMENTED** [review]
Fix minor notes, please (see comments).
Expand cases by new features (see comments as well).

File: docs/testcases/UI/HCS/2489_2758.md (hunk 1 above)
Comment (maryvictol) [reply docs/testcases/UI/HCS/2489_2758.md]: Done

File: docs/testcases/UI/HCS/2527.md (hunk 2 above)
Comment (maryvictol) [reply docs/testcases/UI/HCS/2527.md]: Done

File: docs/testcases/UI/HCS/2527.md (hunk 3 above)
Comment (maryvictol) [reply docs/testcases/UI/HCS/2527.md]: Done

File: docs/testcases/UI/HCS/2694_4.md (hunk 5 above)
Comment (maryvictol) [reply docs/testcases/UI/HCS/2694_4.md]: Done

File: docs/testcases/UI/HCS/2749.md (hunk 6 above)
Comment (maryvictol) [reply docs/testcases/UI/HCS/2749.md]: Done

File: docs/testcases/UI/HCS/2766.md (hunk 7 above)
Comment (maryvictol) [reply docs/testcases/UI/HCS/2766.md]: Done

File: docs/testcases/UI/HCS/2766.md (hunk 8 above)
Comment (maryvictol) [reply docs/testcases/UI/HCS/2766.md]: Done

File: docs/testcases/UI/HCS/2528.md (hunk 4 above)
Comment (maryvictol) [reply docs/testcases/UI/HCS/2528.md]: Done

File: docs/testcases/UI/HCS/2766.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+| 11 | Select 1 field on the *Well* panel | Video for selected field with frames delay specified at step 10 is started on the ***Preview panel*** | 
+| 12 | Change Channels settings | Video for selected field and channels settings and with frames delay specified at step 10 is started on the ***Preview panel*** |
+| 13 | Click **Download current video** button | Video from step 12 is downloaded automatically with name in the format: `<HCS_filename>-r<well_row>c<well_col>[f<field_number>]p<plane_number>ts<time_series_number>.<videofile_format>` <br> Where: <br> `<HCS_filename>` - name of the HCS image <br> `<well_row>` - row number of the selected well <br> `<well_col>` - column number of the selected well <br> `<field_number>` - number of the selected field. If the video is created over the whole well, this part shall be missed <br> `<plane_number>` - number of the selected plane <br> `<time_series_number>` - number of the time sequence by which the video is created |
+| 14 | Reproduce downloaded video in any media player | Downloaded video corresponds video displayed at step 12 on the ***Preview panel*** |
+| 15 | Close ***Preview pop-up*** | |
```
Comment (NShaforostov) [inline docs/testcases/UI/HCS/2766.md]: Fix to `Close ***Settings*** panel`

**Review by NShaforostov — APPROVED** [review]
Fix a minor note (see the comment to 2766.md file) before the merging, please.

File: docs/testcases/UI/HCS/2766.md (hunk 10 above)
Comment (maryvictol) [reply docs/testcases/UI/HCS/2766.md]: Done

**Pull request merged**